lib/vrl/compiler/src/expression/query.rs

```diff
@@ -74,7 +75,7 @@ impl Expression for Query {
                 // `.` path must be an object
                 //
                 // TODO: make sure to enforce this
-                if self.path.is_root() {
+                if self.path.is_empty() {
```

Comment:
Maybe add an alias for `is_root`, since it more clearly describes what's being expected in this condition.